include/boost/geometry/algorithms/detail/point_on_border.hpp

```diff
@@ -273,31 +268,45 @@ struct point_on_border<multi_linestring_tag, Point, Multi>
 \tparam Geometry geometry type. This also defines the type of the output point
 \param point to assign
 \param geometry geometry to take point from
-\param midpoint boolean flag, true if the point should not be a vertex, but some point
-    in between of two vertices
 \return TRUE if successful, else false.
     It is only false if polygon/line have no points
 \note for a polygon, it is always a point on the exterior ring
-\note for take_midpoint, it is not taken from two consecutive duplicate vertices,
-    (unless there are no other).
  */
 template <typename Point, typename Geometry>
-inline bool point_on_border(Point& point,
-            Geometry const& geometry,
-            bool midpoint = false)
+inline bool point_on_border(Point& point, Geometry const& geometry)
 {
     concepts::check<Point>();
     concepts::check<Geometry const>();
 
     return dispatch::point_on_border
             <
                 typename tag<Geometry>::type,
-                Point,
-                Geometry
-            >::apply(point, geometry, midpoint);
+                false
+            >::apply(point, geometry);
 }
 
 
+/*!
+\tparam Midpoint boolean flag, true if the point should not be a vertex, but some point
+    in between of two vertices
+\note for Midpoint, it is not taken from two consecutive duplicate vertices,
+    (unless there are no other).
+ */
+/*
+template <bool Midpoint, typename Point, typename Geometry>
+inline bool point_on_border(Point& point, Geometry const& geometry)
+{
+    concepts::check<Point>();
+    concepts::check<Geometry const>();
+
+    return dispatch::point_on_border
+            <
+                typename tag<Geometry>::type,
+                Midpoint
+            >::apply(point, geometry);
+}
+*/
```

Comment:
This is commented. Maybe we can omit this functionality. Don't think it is ever used, though we can of course not be sure about that. The bool flag was only added for overlay.
For the rest - changes look OK to me, I did not get into the details but I think it is an enhancement indeed. Thanks.